File: merge-two-2d-arrays-by-summing-values/solution.py
class Solution:
    def mergeArrays(self, nums1: List[List[int]], nums2: List[List[int]]) -> List[List[int]]:
        d=defaultdict(int)
        for i,v in nums1+nums2:
            d[i]+=v
        return [[key,d[key]] for key in sorted(d.keys())]

File: merge-two-2d-arrays-by-summing-values/solution.cpp
class Solution {
public:
    vector<vector<int>> mergeArrays(vector<vector<int>>& nums1, vector<vector<int>>& nums2) {
        map<int,int>d;
        for(auto &p:nums1){
            d[p[0]]+=p[1];
        }
        for(auto &p:nums2){
            d[p[0]]+=p[1];
        }
        vector<vector<int>>ans;
        for(auto &i:d)
            ans.push_back({i.first,i.second});
        return ans;
    }
};